Automattic/wp-calypso
# Improve price dropdown to show savings

Related to https://github.com/Automattic/jetpack-marketing/issues/496

## Proposed Changes

* Separate code for Jetpack variation dropdown
* Update discount percentage to the discounted amount
* Update styling in Jetpack checkout

![CleanShot 2024-01-26 at 12 35 42@2x](https://github.com/Automattic/wp-calypso/assets/8419292/788b20d4-891d-44f6-9c5e-0f2985c30adc)


## Testing Instructions

* Go to http://calypso.localhost:3000/checkout/jetpack/jetpack_social_advanced_yearly
* Make sure that dropdown shows the correct sum of discounts broken down on the right side
![CleanShot 2024-01-26 at 12 36 21@2x](https://github.com/Automattic/wp-calypso/assets/8419292/67f0bb9c-cb3b-456e-a924-850c42e2b548)

* Go to http://calypso.localhost:3000/checkout/jetpack/jetpack_complete
* Again here, make sure that the sum in dropdown is the sae as in the price breakdown
![CleanShot 2024-01-26 at 12 37 41@2x](https://github.com/Automattic/wp-calypso/assets/8419292/ca8855e5-cbaa-49bf-ad10-d1464ca82958)


## Pre-merge Checklist
